### PR TITLE
Introduce testing-library

### PR DESCRIPTION
**Pull request by simonrelet**
## Description

Support [React Testing Library](https://testing-library.com/docs/react-testing-library/intro) in our tests.

## What has been done

- Update Jest config
- Add ESLint recommended rules for RTL and jest-dom
- Write `HeroSection` and `UneditableTextField` tests with RTL.

## How it was tested

With the unit tests 🤷‍♂️

### CODE REVIEW

**Comment by mcellier** [comment]
Can we have an example with snapshots too?

**Comment by simonrelet** [comment]
> @mcellier 
> * Can we have an example with snapshots too?

I didn't find a good use case for snapshot testing, if you have one I can update it too.

> * (Maybe) add an entry to the changelog

I didn't update it because this PR doesn't changes anything from the developer user point of view, i.e. it doesn't change the production code.

**Comment by mcellier** [comment]
> > @mcellier
> > 
> > * Can we have an example with snapshots too?
> 
> I didn't find a good use case for snapshot testing, if you have one I can update it too.

Simple usage: `UneditableTextField`

> > * (Maybe) add an entry to the changelog
> 
> I didn't update it because this PR doesn't changes anything from the developer user point of view, i.e. it doesn't change the production code.

We're pretty inconsistent on that part, but I agree with you.

**Comment by simonrelet** [comment]
> @mcellier
> Simple usage: `UneditableTextField`

Done in this PR, snapshots were not needed 😉.

File: src/uneditableTextField/index.unit.tsx [hunk 1]
```diff
@@ -1,41 +1,74 @@
...
-    const wrapper = renderer.create(<UneditableTextField>Hello world</UneditableTextField>)
-    expect(wrapper.toJSON()).toMatchSnapshot()
+  it('should render the content', () => {
+    const props = createProps()
+    const { container } = render(<UneditableTextField {...props} />)
+
+    expect(container.firstChild).toHaveTextContent(props.children as string)
+  })
+
+  it('should apply public CSS classes', () => {
```
Comment (simonrelet) [inline src/uneditableTextField/index.unit.tsx:19]: I not sure we should have CSS classes in the public API of a component but they are used so I added a specific test to be isofunctional.

File: src/uneditableTextField/index.unit.tsx [hunk 2]
```diff
@@ -1,41 +1,74 @@
...
   })
 
-  it('Should have the text ellipsed', () => {
-    const wrapper = renderer.create(
-      <UneditableTextField ellipsis>
-        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec nec tristique sapien, eu
-        placerat justo. Donec tempor, risus ac cursus fringilla, lorem ipsum facilisis tortor, vel
-        molestie sapien justo nec orci.
-      </UneditableTextField>,
+  it('Should not add CSS class for text ellipsis by default', () => {
```
Comment (simonrelet) [inline src/uneditableTextField/index.unit.tsx:27]: A screenshot test would be the best here.

**Pull request merged**